tests/e2e/vmservice_late_binding.go

```diff
@@ -269,4 +309,563 @@ var _ bool = ginkgo.Describe("[vmsvc] VM-Service-VM-LateBinding", func() {
 
 	})
 
+	/**
+	1. Create multiple PVC's . Consider 2 PVC's with WFFC and 2 PVC's with immediate
+	2. Use the above  PVC's and create a VmService VM's
+	3. Wait for VM  get powered ON
+	4. Once the VM is on Verify that the PVC's with WFFC should go to bound state
+		Describe PVC and verify the below annotations
+		cns.vmware.com/selected-node-is-zone is set to true
+		volume.kubernetes.io/selected-node is set to  Zone - This should have the zone name where
+		the VM gets provisioned
+	5. PVC's with Immediate binding will not have any annotation
+	6. Verify CNS metadata for PVC
+	7. Verify the storagePolicyQuotaUsage and VMstorageQuotaUsage CR's on the late binding storage class
+	   Quota should have the appropriate quota consumption details
+	   Clean up all the above data
+	*/
+	ginkgo.It("MultiplePVC-AttachedTo-SingleVM-ImmediateAndWFFC", ginkgo.Label(p0, block, wcp, vmsvc, vc901), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Get WFFC stotage class")
+		wffcStorageclass, err := client.StorageV1().StorageClasses().Get(ctx, wffsStoragePolicyName,
+			metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("wffsStoragePolicyName: %s wffcStorageclass: %s", wffsStoragePolicyName, wffcStorageclass)
+
+		immediateBindingStorageclass, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyName,
+			metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("wffsStoragePolicyName: %s wffcStorageclass: %s", wffsStoragePolicyName, wffcStorageclass)
+
+		ginkgo.By("Create a PVC")
+		pvc1, err := createPVC(ctx, client, namespace, nil, "", wffcStorageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create a PVC")
+		pvc2, err := createPVC(ctx, client, namespace, nil, "", wffcStorageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create a PVC")
+		pvc3, err := createPVC(ctx, client, namespace, nil, "", immediateBindingStorageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create a PVC")
+		pvc4, err := createPVC(ctx, client, namespace, nil, "", immediateBindingStorageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Wait for VM images to get listed under namespace and create VM")
+		err = pollWaitForVMImageToSync(ctx, namespace, vmImageName, poll, supervisorClusterOperationsTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Creating VM bootstrap data")
+		secretName := createBootstrapSecretForVmsvcVms(ctx, client, namespace)
+
+		ginkgo.By("Create vm service vm")
+		vm1 := createVmServiceVmWithPvcs(
+			ctx, vmopC, namespace, vmClass, []*v1.PersistentVolumeClaim{pvc1, pvc2, pvc3, pvc4}, vmi,
+			wffsStoragePolicyName, secretName)
+
+		ginkgo.By("Creating loadbalancing service for ssh with the VM")
```

Comment:
Instead of doing all these steps, please use this util -> createVmServiceVm

It will take care of complete VM Service VM creation

Reply:
Updated.
using createVmServiceVmV4